File: app/src/main/java/com/theone/music/data/room/DownloadDao.java
package com.theone.music.data.room;//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.theone.music.data.model.Download;
import com.theone.music.data.model.User;

import java.util.List;

/**
 * @author The one
 * @date 2022-04-12 10:16
 * @describe Download_Dao
 * @email 625805189@qq.com
 * @remark
 */
@Dao
public interface DownloadDao {

    /**
     * 插入一条数据
     * @param download
     */
    @Insert
    void insert(@NonNull Download download);

    /**
     * 更新数据
     * @param download
     * @return
     */
    @Update
    int update(@NonNull Download download);


    /**
     * 删除一条数据
     * @param download
     * @return
     */
    @Delete
    int delete(@NonNull Download download);


    /**
     * 查询下载记录
     * @param page
     * @param pageSize
     * @return
     */
    @Query("select * from download order by time desc limit :pageSize offset (:page -1)*:pageSize")
    List<Download> getDownloadList(int page,int pageSize);

    /**
     * 更新下载状态
     * @param status 下载状态 [DownloadStatus]
     * @param musicId 下载的音乐本地保存的id
     * @return 更新结果
     */
    @Query("update download set status =:status where musicId = :musicId ")
    int updateDownloadStatus(int status,int musicId);

}

File: app/src/main/java/com/theone/music/data/room/DownloadDao.kt
package com.theone.music.data.room

import androidx.room.*
import com.theone.music.data.model.Download

//  ┏┓　　　┏┓
//┏┛┻━━━┛┻┓
//┃　　　　　　　┃
//┃　　　━　　　┃
//┃　┳┛　┗┳　┃
//┃　　　　　　　┃
//┃　　　┻　　　┃
//┃　　　　　　　┃
//┗━┓　　　┏━┛
//    ┃　　　┃                  神兽保佑
//    ┃　　　┃                  永无BUG！
//    ┃　　　┗━━━┓
//    ┃　　　　　　　┣┓
//    ┃　　　　　　　┏┛
//    ┗┓┓┏━┳┓┏┛
//      ┃┫┫　┃┫┫
//      ┗┻┛　┗┻┛
/**
 * @author The one
 * @date 2021-10-08 09:31
 * @describe TODO
 * @email 625805189@qq.com
 * @remark
 */
@Dao
interface DownloadDao {

    @Insert
    fun insert(bean: Download)

    @Update
    fun update(bean: Download): Int

    @Delete
    fun delete(bean: Download): Int

    @Query("select * from download order by time desc limit :pageSize offset (:page -1)*:pageSize")
    fun getDownloadList(page:Int,pageSize:Int): List<Download>

    @Query("update download set status =:status where musicId = :musicId ")
    fun updateDownloadStatus(status:Int,musicId:Int):Int

}